src/message.c

```diff
@@ -846,6 +853,8 @@ qd_message_t *qd_message()
     }
 
     ZERO(msg->content);
+    msg->content->receive_complete = false;
+    msg->content->discard          = false;
```

Comment:
Agreed. Calling ZERO(msg->content) will set those flags to 0 (false) already. So, why bother setting them to false again. I will remove this.